AdyenCashAppPay/CashAppPayComponent.swift

```diff
@@ -0,0 +1,260 @@
+        formViewController.title = paymentMethod.displayInformation(using: configuration.localizationParameters).title
+    
+        if configuration.showsStorePaymentMethodField {
+            formViewController.append(storeDetailsItem)
+        }
+    
+        formViewController.append(FormSpacerItem(numberOfSpaces: 2))
+        formViewController.append(cashAppPayButton)
+
+        return formViewController
+    }()
+
+    /// Initializes the Cash App Pay component.
+    ///
+    /// - Parameter paymentMethod: The Cash App Pay  payment method.
+    /// - Parameter context: The context object for this component.
+    /// - Parameter configuration: The configuration for the component.
+    public init(paymentMethod: CashAppPayPaymentMethod,
+                context: AdyenContext,
+                configuration: CashAppPayConfiguration) {
+        self.cashAppPayPaymentMethod = paymentMethod
+        self.context = context
+        self.configuration = configuration
+    }
+
+    private func didSelectSubmitButton() {
+        guard formViewController.validate() else { return }
+    
+        startLoading()
+        startCashAppPayFlow()
+    }
+
+    private func startLoading() {
+        cashAppPayButton.showsActivityIndicator = true
+        formViewController.view.isUserInteractionEnabled = false
+    }
+
+    public func stopLoading() {
+        cashAppPayButton.showsActivityIndicator = false
+        formViewController.view.isUserInteractionEnabled = true
+    }
+
+    private func startCashAppPayFlow() {
+        let actions = createPaymentActions()
+        guard actions.isEmpty == false else {
+            AdyenAssertion.assertionFailure(message: "At least one paymentAction is required to create a CashAppPay rquest")
+            return
+        }
+    
+        cashAppPay.createCustomerRequest(params: CreateCustomerRequestParams(actions: actions,
+                                                                             redirectURL: configuration.redirectURL,
+                                                                             referenceID: configuration.referenceId,
+                                                                             metadata: nil))
+    }
+
+    private func createPaymentActions() -> [PaymentAction] {
+        var actions = [PaymentAction]()
+        if let amount = payment?.amount, amount.value > 0 {
+            let moneyAmount = Money(amount: UInt(amount.value), currency: .USD)
+            let oneTimeAction = PaymentAction.oneTimePayment(scopeID: cashAppPayPaymentMethod.scopeId,
+                                                             money: moneyAmount)
+            actions.append(oneTimeAction)
+        }
+    
+        if storePayment {
+            let onFileAction = PaymentAction.onFilePayment(scopeID: cashAppPayPaymentMethod.scopeId,
+                                                           accountReferenceID: nil)
```

Comment:
Here should be `referenceId` ?

Reply:
No referenceId and accountReferenceId are different things on cashapp